app/controllers/media_controller.rb

```diff
@@ -19,14 +19,24 @@ def stream
     authorize! :read, @media
     respond_to do |format|
       format.m3u8 do
-        redirect_to @media.to_playlist_url
+        redirect_to "#{@media.to_playlist_url}?token_string=#{media_token}"
       end
       format.mpd do
-        redirect_to @media.to_manifest_url
+        redirect_to "#{@media.to_manifest_url}?token_string=#{media_token}"
       end
     end
   end
 
+  # rubocop:disable Metrics/LineLength
+  def verify_token
+    if StacksMediaToken.verify_encrypted_token_string params[:token_string], params[:id], params[:file_name], params[:user_ip_addr]
```

Comment:
sure, that seems reasonable.

i figured the `allowed_` part would get added when i rebased on master after the strong params PR got merged.